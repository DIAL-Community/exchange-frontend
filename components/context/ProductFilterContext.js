import { createContext, useState } from 'react'

const ProductFilterContext = createContext()
const ProductFilterDispatchContext = createContext()

const ProductFilterProvider = ({ children }) => {
  const [isEndorsed, setIsEndorsed] = useState(false)
  const [productDeployable, setProductDeployable] = useState(false)
  const [origins, setOrigins] = useState([])
  const [countries, setCountries] = useState([])
  const [sectors, setSectors] = useState([])
  const [organizations, setOrganizations] = useState([])
  const [sdgs, setSdgs] = useState([])
  const [tags, setTags] = useState([])
  const [useCases, setUseCases] = useState([])
  const [workflows, setWorkflows] = useState([])
  const [buildingBlocks, setBuildingBlocks] = useState([])
  const [endorsers, setEndorsers] = useState([])
  const [licenseTypes, setLicenseTypes] = useState([])
  const [isLinkedWithDpi, setIsLinkedWithDpi] = useState(false)
  const [showGovStackOnly, setShowGovStackOnly] = useState(false)
  const [showDpgaOnly, setShowDpgaOnly] = useState(false)

  const [softwareCategories, setSoftwareCategories] = useState([])
  const [softwareFeatures, setSoftwareFeatures] = useState([])
  const [comparedProducts, setComparedProducts] = useState([])

  const [search, setSearch] = useState('')

  const [pageNumber, setPageNumber] = useState(0)
  const [pageOffset, setPageOffset] = useState(0)

  const productFilterValues = {
    isEndorsed,
    productDeployable,
    origins,
    countries,
    sectors,
    organizations,
    sdgs,
    tags,
    useCases,
    workflows,
    buildingBlocks,
    endorsers,
    licenseTypes,
    isLinkedWithDpi,
    showGovStackOnly,
    showDpgaOnly,
    softwareCategories,
    softwareFeatures,
    comparedProducts,
    search,
    pageOffset,
    pageNumber
  }

  const productFilterDispatchValues = {
    setIsEndorsed,
    setProductDeployable,
    setOrigins,
    setCountries,
    setSectors,
    setOrganizations,
    setSdgs,
    setTags,
    setUseCases,
    setWorkflows,
    setBuildingBlocks,
    setEndorsers,
    setLicenseTypes,
    setIsLinkedWithDpi,
    setShowGovStackOnly,
    setShowDpgaOnly,
    setSoftwareCategories,
    setSoftwareFeatures,
    setComparedProducts,
    setSearch,
    setPageNumber,
    setPageOffset
  }

  return (
    <ProductFilterContext.Provider value={productFilterValues}>
      <ProductFilterDispatchContext.Provider value={productFilterDispatchValues}>
        {children}
      </ProductFilterDispatchContext.Provider>
    </ProductFilterContext.Provider>
  )
}

export { ProductFilterProvider, ProductFilterContext, ProductFilterDispatchContext }
