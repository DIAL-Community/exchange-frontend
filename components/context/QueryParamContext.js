import { createContext, useState } from 'react'

const QueryParamContext = createContext()
const QueryParamContextProvider = ({ children }) => {
  const [interactionDetected, setInteractionDetected] = useState(false)
  return (
    <QueryParamContext.Provider value={{ ...{ interactionDetected, setInteractionDetected } }}>
      {children}
    </QueryParamContext.Provider>
  )
}

export { QueryParamContextProvider, QueryParamContext }
