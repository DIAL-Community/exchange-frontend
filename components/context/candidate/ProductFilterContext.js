import React, { createContext, useState } from 'react'

const ProductFilterContext = createContext()
const ProductFilterDispatchContext = createContext()

const ProductFilterProvider = ({ children }) => {
  const [search, setSearch] = useState('')

  const candidateProductFilterValues = {
    search
  }
  const candidateProductFilterDispatchValues = {
    setSearch
  }

  return (
    <ProductFilterContext.Provider value={candidateProductFilterValues}>
      <ProductFilterDispatchContext.Provider value={candidateProductFilterDispatchValues}>
        {children}
      </ProductFilterDispatchContext.Provider>
    </ProductFilterContext.Provider>
  )
}

export { ProductFilterProvider, ProductFilterContext, ProductFilterDispatchContext }
