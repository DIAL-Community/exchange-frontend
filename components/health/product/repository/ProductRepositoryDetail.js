import { useRef, useState } from 'react'
import { useApolloClient, useQuery } from '@apollo/client'
import { GRAPH_QUERY_CONTEXT } from '../../../../lib/apolloClient'
import { handleLoadingQuery, handleMissingData, handleQueryError } from '../../../shared/GraphQueryHandler'
import { PRODUCT_POLICY_QUERY } from '../../../shared/query/product'
import { PRODUCT_REPOSITORY_DETAIL_QUERY } from '../../../shared/query/productRepository'
import { fetchOperationPolicies } from '../../../utils/policy'
import Breadcrumb from '../../shared/Breadcrumb'
import ProductRepositoryDetailLeft from './ProductRepositoryDetailLeft'
import ProductRepositoryDetailRight from './ProductRepositoryDetailRight'

const ProductRepositoryDetail = ({ productSlug, repositorySlug }) => {
  const scrollRef = useRef(null)
  const client = useApolloClient()

  const [editingAllowed, setEditingAllowed] = useState(false)

  const { loading, error, data } = useQuery(PRODUCT_REPOSITORY_DETAIL_QUERY, {
    variables: { productSlug, repositorySlug },
    context: {
      headers: {
        ...GRAPH_QUERY_CONTEXT.VIEWING
      }
    }
  })

  if (loading) {
    return handleLoadingQuery
  } else if (error) {
    return handleQueryError(error)
  } else if (!data?.productRepository || !data?.product) {
    return handleMissingData()
  }

  const { product, productRepository } = data

  fetchOperationPolicies(
    client,
    PRODUCT_POLICY_QUERY,
    ['editing']
  ).then(policies => {
    setEditingAllowed(policies['editing'])
  })

  const slugNameMapping = (() => {
    const map = {}
    map[product.slug] = product.name
    map[productRepository.slug] = productRepository.name

    return map
  })()

  return (
    <div className='lg:px-8 xl:px-56 flex flex-col'>
      <div className="py-6 text-dial-stratos z-40">
        <Breadcrumb slugNameMapping={slugNameMapping}/>
        <hr className="border-b border-health-gray my-3"/>
      </div>
      <div className='flex flex-col lg:flex-row gap-x-8'>
        <div className='lg:basis-1/3'>
          <ProductRepositoryDetailLeft
            scrollRef={scrollRef}
            product={product}
            productRepository={productRepository}
          />
        </div>
        <div className='lg:basis-2/3'>
          <ProductRepositoryDetailRight
            ref={scrollRef}
            product={product}
            productRepository={productRepository}
            editingAllowed={editingAllowed}
          />
        </div>
      </div>
    </div>
  )
}

export default ProductRepositoryDetail
