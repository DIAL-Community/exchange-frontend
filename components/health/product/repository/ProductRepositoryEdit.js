import { useCallback, useRef } from 'react'
import { useIntl } from 'react-intl'
import { useQuery } from '@apollo/client'
import { PRODUCT_REPOSITORY_DETAIL_QUERY } from '../../../shared/query/productRepository'
import Breadcrumb from '../../shared/Breadcrumb'
import { Error, Loading, NotFound } from '../../../shared/FetchStatus'
import ProductRepositoryForm from './fragments/ProductRepositoryForm'
import ProductRepositoryEditLeft from './ProductRepositoryEditLeft'

const ProductRepositoryEdit = ({ productSlug, repositorySlug }) => {
  const scrollRef = useRef(null)

  const { formatMessage } = useIntl()
  const format = useCallback((id, values) => formatMessage({ id }, values), [formatMessage])

  const { loading, error, data } = useQuery(PRODUCT_REPOSITORY_DETAIL_QUERY, {
    variables: { productSlug, repositorySlug }
  })

  if (loading) {
    return <Loading />
  } else if (error) {
    return <Error />
  } else if (!data?.product) {
    return <NotFound />
  }

  const { product, productRepository } = data

  const slugNameMapping = (() => {
    const map = {
      edit: format('app.edit')
    }
    map[product.slug] = product.name
    map[productRepository.slug] = productRepository.name

    return map
  })()

  return (
    <div className='lg:px-8 xl:px-56 flex flex-col'>
      <div className='px-4 lg:px-6 py-4 bg-dial-blue-chalk text-dial-stratos ribbon-detail z-40'>
        <Breadcrumb slugNameMapping={slugNameMapping}/>
      </div>
      <div className='flex flex-col lg:flex-row gap-x-8'>
        <div className='lg:basis-1/3'>
          <ProductRepositoryEditLeft product={product} scrollRef={scrollRef} />
        </div>
        <div className='lg:basis-2/3'>
          <ProductRepositoryForm
            ref={scrollRef}
            product={product}
            productRepository={productRepository}
          />
        </div>
      </div>
    </div>
  )
}

export default ProductRepositoryEdit
