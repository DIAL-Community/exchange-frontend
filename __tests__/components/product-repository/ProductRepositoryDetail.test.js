import { screen } from '@testing-library/dom'
import userEvent from '@testing-library/user-event'
import { ProductFilterProvider } from '../../../components/context/ProductFilterContext'
import { QueryParamContextProvider } from '../../../components/context/QueryParamContext'
import ProductRepositoryDetail from '../../../components/product/repository/ProductRepositoryDetail'
import ProductRepositoryEdit from '../../../components/product/repository/ProductRepositoryEdit'
import { CREATE_PRODUCT_REPOSITORY } from '../../../components/shared/mutation/productRepository'
import { OWNED_PRODUCTS_QUERY } from '../../../components/shared/query/product'
import {
  PRODUCT_REPOSITORIES_QUERY, PRODUCT_REPOSITORY_DETAIL_QUERY
} from '../../../components/shared/query/productRepository'
import { render } from '../../test-utils'
import CustomMockedProvider, { generateMockApolloData } from '../../utils/CustomMockedProvider'
import { mockNextAuthUseSession, mockNextUseRouter, mockTenantApi } from '../../utils/nextMockImplementation'
import {
  createProductRepository, ownedProducts, productRepositories, productRepositoryDetail
} from './data/ProductRepositoryDetail.data'

mockTenantApi()
mockNextUseRouter()
describe('Unit tests for the product main page.', () => {
  const mockProductRepositoryDetail = generateMockApolloData(
    PRODUCT_REPOSITORY_DETAIL_QUERY,
    {
      'productSlug': 'firma',
      'repositorySlug': 'firma-repository'
    },
    null,
    productRepositoryDetail
  )

  const mockOwnedProducts = generateMockApolloData(
    OWNED_PRODUCTS_QUERY,
    {},
    null,
    ownedProducts
  )

  const mockProductRepositories = generateMockApolloData(
    PRODUCT_REPOSITORIES_QUERY,
    { 'productSlug': 'firma' },
    null,
    productRepositories
  )

  test('Should render the product repository page.', async () => {

    const { container } = render(
      <CustomMockedProvider mocks={[mockOwnedProducts, mockProductRepositories, mockProductRepositoryDetail]}>
        <QueryParamContextProvider>
          <ProductFilterProvider>
            <ProductRepositoryDetail productSlug={'firma'} repositorySlug={'firma-repository'} />
          </ProductFilterProvider>
        </QueryParamContextProvider>
      </CustomMockedProvider>
    )

    expect(await screen.findByText('@firma')).toBeInTheDocument()
    expect(await screen.findByText('Repository of @firma.')).toBeInTheDocument()
    expect(await screen.findByText('administracionelectronica.gob.es/ctt/clienteafirma')).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  test('Should render unauthorized to edit repository.', async () => {
    const { container } = render(
      <CustomMockedProvider mocks={[mockOwnedProducts, mockProductRepositories, mockProductRepositoryDetail]}>
        <QueryParamContextProvider>
          <ProductFilterProvider>
            <ProductRepositoryEdit productSlug={'firma'} repositorySlug={'firma-repository'} />
          </ProductFilterProvider>
        </QueryParamContextProvider>
      </CustomMockedProvider>
    )

    expect(await screen.findByText('@firma')).toBeInTheDocument()
    expect(await screen.findByText('You are not authorized to view this page')).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  test('Should render the product repository edit page.', async () => {
    // Set auth value, mock that user is authenticated
    mockNextAuthUseSession()

    const mockCreateProductRepository = generateMockApolloData(
      CREATE_PRODUCT_REPOSITORY,
      {
        'productSlug':'firma',
        'name':'@firma Repository Information',
        'slug':'firma-repository',
        'absoluteUrl':'github.com/ctt-gob-es/clienteafirma',
        'description':'Repository of @firma.',
        'mainRepository':true
      },
      null,
      createProductRepository
    )

    const { container } = render(
      <CustomMockedProvider
        mocks={[
          mockOwnedProducts,
          mockProductRepositories,
          mockProductRepositoryDetail,
          mockCreateProductRepository
        ]}
      >
        <QueryParamContextProvider>
          <ProductFilterProvider>
            <ProductRepositoryEdit productSlug={'firma'} repositorySlug={'firma-repository'} />
          </ProductFilterProvider>
        </QueryParamContextProvider>
      </CustomMockedProvider>
    )

    expect(await screen.findByText('@firma')).toBeInTheDocument()
    expect(await screen.findByText('administracionelectronica.gob.es/ctt/clienteafirma')).toBeInTheDocument()

    const repositoryNameInput = screen.getByDisplayValue('@firma Repository')
    expect(repositoryNameInput.value).toBe('@firma Repository')

    const user = userEvent.setup()
    await user.type(repositoryNameInput, ' Information')
    expect(repositoryNameInput.value).toBe('@firma Repository Information')

    const repositorySubmitButton = screen.getByText('Submit Product')
    await user.click(repositorySubmitButton)

    expect(container).toMatchSnapshot()
  })
})
