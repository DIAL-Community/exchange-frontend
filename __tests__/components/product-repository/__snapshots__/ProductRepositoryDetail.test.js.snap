// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit tests for the product main page. Should render the product repository edit page. 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="lg:px-8 xl:px-56 flex flex-col"
  >
    <div
      class="px-4 lg:px-6 py-4 bg-dial-blue-chalk text-dial-stratos ribbon-detail z-40"
    >
      <div
        class="whitespace-nowrap text-ellipsis overflow-hidden"
      >
        <a
          class="h5"
          href="/"
        >
          Home
        </a>
      </div>
    </div>
    <div
      class="flex flex-col lg:flex-row gap-x-8"
    >
      <div
        class="lg:basis-1/3"
      >
        <div
          class="bg-dial-slate-100 lg:h-full"
        >
          <div
            class="flex flex-col gap-y-3 px-4 lg:px-6 lg:py-3"
          >
            <div
              class="flex flex-col gap-y-4 py-3"
            >
              <div
                class="text-xl text-dial-meadow font-semibold"
              >
                @firma
              </div>
              <div
                class="flex justify-center items-center py-16 bg-white rounded border"
              >
                <div
                  class="inline"
                >
                  <img
                    alt="Logo of Product."
                    class="object-contain"
                    src="http://localhost:3000//assets/products/firma.png"
                  />
                </div>
              </div>
              <div
                class="flex flex-col gap-y-8 text-sm pt-6 pb-3"
              >
                <div
                  class="flex flex-col gap-y-3"
                >
                  <div
                    class="font-semibold text-dial-meadow"
                  >
                    Website
                  </div>
                  <div
                    class="flex text-dial-stratos"
                  >
                    <a
                      class="flex border-b border-dial-iris-blue "
                      href="//administracionelectronica.gob.es/ctt/clienteafirma"
                      rel="noreferrer"
                      target="_blank"
                    >
                      <div
                        class="line-clamp-1 break-all"
                      >
                        administracionelectronica.gob.es/ctt/clienteafirma
                      </div>
                    </a>
                     ⧉
                  </div>
                </div>
                <div
                  class="flex flex-col gap-y-3"
                >
                  <div
                    class="font-semibold text-dial-meadow"
                  >
                    License
                  </div>
                  <div
                    class="flex text-dial-stratos"
                  >
                    NONE
                  </div>
                </div>
                <div
                  class="flex flex-col gap-y-3"
                >
                  <div
                    class="flex flex-row gap-3"
                  >
                    <div
                      class="font-semibold text-dial-meadow"
                    >
                      Sectors
                    </div>
                    <div
                      class="flex gap-3 ml-auto"
                    />
                  </div>
                  <div
                    class="flex flex-col gap-y-2"
                  >
                    <div>
                      <div>
                        State and Democracy (Good Governance)
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <hr
              class="border-b border-dial-slate-200"
            />
            <div
              class="flex flex-col gap-y-3 text-sm py-3"
            >
              <div
                class="font-semibold text-dial-blueberry"
              >
                Jump To
              </div>
              <div
                class=" css-1uaxf62-container"
              >
                <span
                  class="css-1f43avz-a11yText-A11yText"
                  id="react-select-:r1:-live-region"
                />
                <span
                  aria-atomic="false"
                  aria-live="polite"
                  aria-relevant="additions text"
                  class="css-1f43avz-a11yText-A11yText"
                />
                <div
                  class="react-select__control css-afrqkl-control"
                >
                  <div
                    class="react-select__value-container css-17t7gkf-ValueContainer"
                  >
                    <div
                      class="react-select__placeholder css-5rjclb-placeholder"
                      id="react-select-:r1:-placeholder"
                    >
                      Select...
                    </div>
                    <div
                      class="react-select__input-container css-qbdosj-Input"
                      data-value=""
                    >
                      <input
                        aria-autocomplete="list"
                        aria-describedby="react-select-:r1:-placeholder"
                        aria-expanded="false"
                        aria-haspopup="true"
                        aria-label="Current active navigation"
                        autocapitalize="none"
                        autocomplete="off"
                        autocorrect="off"
                        class="react-select__input"
                        id="async-select-id-undefined"
                        role="combobox"
                        spellcheck="false"
                        style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                        tabindex="0"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
                  >
                    <span
                      class="react-select__indicator-separator css-nyxz9h-indicatorSeparator"
                    />
                    <div
                      aria-hidden="true"
                      class="react-select__indicator react-select__dropdown-indicator css-2hgo35-indicatorContainer"
                    >
                      <svg
                        aria-hidden="true"
                        class="css-tj5bde-Svg"
                        focusable="false"
                        height="20"
                        viewBox="0 0 20 20"
                        width="20"
                      >
                        <path
                          d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="lg:basis-2/3"
      >
        <form>
          <div
            class="px-4 lg:px-0 py-4 lg:py-6 text-dial-meadow"
          >
            <div
              class="flex flex-col gap-y-6 text-sm"
            >
              <div
                class="text-xl font-semibold"
              >
                Edit @firma
              </div>
              <div
                class="flex flex-col gap-y-2"
              >
                <label
                  class="required-field"
                  for="name"
                >
                  Repository Name
                </label>
                <input
                  class="w-full text-sm"
                  id="name"
                  name="name"
                  placeholder="Repository Name"
                  type="text"
                  value=""
                />
              </div>
              <label
                class="flex gap-x-2 items-center self-start text-dial-sapphire"
              >
                <input
                  class="h-5 w-5"
                  name="mainRepository"
                  type="checkbox"
                />
                This is the main repository.
              </label>
              <div
                class="flex flex-col gap-y-2"
              >
                <label
                  for="absoluteUrl"
                >
                  Repository Location
                </label>
                <div
                  class="border border-transparent url-input bg-white flex items-center"
                >
                  <span
                    class="select-none text-dial-gray"
                  >
                    https://
                  </span>
                  <input
                    class="url w-full text-sm placeholder-dial-gray border-0 py-0 px-0 focus:outline-0"
                    id="absoluteUrl"
                    placeholder="Repository Location"
                    value="github.com/ctt-gob-es/clienteafirma"
                  />
                </div>
              </div>
              <div
                class="flex flex-col gap-y-2"
              >
                <label
                  class="required-field"
                >
                  Repository Description
                </label>
                <div
                  class="relative"
                >
                  <svg
                    class="absolute text-lg inset-x-1/2 top-10 spinner"
                    fill="currentColor"
                    height="2em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 512 512"
                    width="2em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M304 48a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zm0 416a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zM48 304a48 48 0 1 0 0-96 48 48 0 1 0 0 96zm464-48a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zM142.9 437A48 48 0 1 0 75 369.1 48 48 0 1 0 142.9 437zm0-294.2A48 48 0 1 0 75 75a48 48 0 1 0 67.9 67.9zM369.1 437A48 48 0 1 0 437 369.1 48 48 0 1 0 369.1 437z"
                    />
                  </svg>
                  <div
                    class="html-editor"
                  >
                    <textarea
                      id="description-editor"
                      style="visibility: hidden;"
                    />
                  </div>
                </div>
              </div>
              <div
                class="flex flex-wrap text-base mt-6 gap-3"
              >
                <button
                  class="submit-button"
                  disabled=""
                  type="submit"
                >
                  Submit Product
                  <svg
                    class="spinner ml-3"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M304 48a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zm0 416a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zM48 304a48 48 0 1 0 0-96 48 48 0 1 0 0 96zm464-48a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zM142.9 437A48 48 0 1 0 75 369.1 48 48 0 1 0 142.9 437zm0-294.2A48 48 0 1 0 75 75a48 48 0 1 0 67.9 67.9zM369.1 437A48 48 0 1 0 437 369.1 48 48 0 1 0 369.1 437z"
                    />
                  </svg>
                </button>
                <button
                  class="cancel-button"
                  disabled=""
                  type="button"
                >
                  Cancel
                </button>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`Unit tests for the product main page. Should render the product repository page. 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="lg:px-8 xl:px-56 flex flex-col"
  >
    <div
      class="px-4 lg:px-6 py-4 bg-dial-blue-chalk text-dial-stratos ribbon-detail z-40"
    >
      <div
        class="whitespace-nowrap text-ellipsis overflow-hidden"
      >
        <a
          class="h5"
          href="/"
        >
          Home
        </a>
      </div>
    </div>
    <div
      class="flex flex-col lg:flex-row gap-x-8"
    >
      <div
        class="lg:basis-1/3"
      >
        <div
          class="bg-dial-slate-100 h-full"
        >
          <div
            class="flex flex-col gap-y-3 px-4 lg:px-6 lg:py-3"
          >
            <div
              class="flex flex-col gap-y-4 py-3"
            >
              <div
                class="text-xl text-dial-meadow font-semibold"
              >
                @firma
              </div>
              <div
                class="flex justify-center items-center py-16 bg-white rounded border"
              >
                <div
                  class="inline"
                >
                  <img
                    alt="Logo of Product."
                    class="object-contain"
                    src="http://localhost:3000//assets/products/firma.png"
                  />
                </div>
              </div>
              <div
                class="flex flex-col gap-y-8 text-sm pt-6 pb-3"
              >
                <div
                  class="flex flex-col gap-y-3"
                >
                  <div
                    class="font-semibold text-dial-meadow"
                  >
                    Website
                  </div>
                  <div
                    class="flex text-dial-stratos"
                  >
                    <a
                      class="flex border-b border-dial-iris-blue "
                      href="//administracionelectronica.gob.es/ctt/clienteafirma"
                      rel="noreferrer"
                      target="_blank"
                    >
                      <div
                        class="line-clamp-1 break-all"
                      >
                        administracionelectronica.gob.es/ctt/clienteafirma
                      </div>
                    </a>
                     ⧉
                  </div>
                </div>
                <div
                  class="flex flex-col gap-y-3"
                >
                  <div
                    class="font-semibold text-dial-meadow"
                  >
                    License
                  </div>
                  <div
                    class="flex text-dial-stratos"
                  >
                    NONE
                  </div>
                </div>
                <div
                  class="flex flex-col gap-y-3"
                >
                  <div
                    class="flex flex-row gap-3"
                  >
                    <div
                      class="font-semibold text-dial-meadow"
                    >
                      Sectors
                    </div>
                    <div
                      class="flex gap-3 ml-auto"
                    />
                  </div>
                  <div
                    class="flex flex-col gap-y-2"
                  >
                    <div>
                      <div>
                        State and Democracy (Good Governance)
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <hr
              class="border-b border-dial-slate-200"
            />
            <div
              class="flex flex-col gap-y-3 text-sm py-3"
            >
              <div
                class="font-semibold text-dial-blueberry"
              >
                Jump To
              </div>
              <div
                class=" css-1uaxf62-container"
              >
                <span
                  class="css-1f43avz-a11yText-A11yText"
                  id="react-select-:r0:-live-region"
                />
                <span
                  aria-atomic="false"
                  aria-live="polite"
                  aria-relevant="additions text"
                  class="css-1f43avz-a11yText-A11yText"
                />
                <div
                  class="react-select__control css-afrqkl-control"
                >
                  <div
                    class="react-select__value-container css-17t7gkf-ValueContainer"
                  >
                    <div
                      class="react-select__placeholder css-5rjclb-placeholder"
                      id="react-select-:r0:-placeholder"
                    >
                      Select...
                    </div>
                    <div
                      class="react-select__input-container css-qbdosj-Input"
                      data-value=""
                    >
                      <input
                        aria-autocomplete="list"
                        aria-describedby="react-select-:r0:-placeholder"
                        aria-expanded="false"
                        aria-haspopup="true"
                        aria-label="Current active navigation"
                        autocapitalize="none"
                        autocomplete="off"
                        autocorrect="off"
                        class="react-select__input"
                        id="async-select-id-undefined"
                        role="combobox"
                        spellcheck="false"
                        style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                        tabindex="0"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
                  >
                    <span
                      class="react-select__indicator-separator css-nyxz9h-indicatorSeparator"
                    />
                    <div
                      aria-hidden="true"
                      class="react-select__indicator react-select__dropdown-indicator css-2hgo35-indicatorContainer"
                    >
                      <svg
                        aria-hidden="true"
                        class="css-tj5bde-Svg"
                        focusable="false"
                        height="20"
                        viewBox="0 0 20 20"
                        width="20"
                      >
                        <path
                          d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <hr
              class="border-b border-dial-slate-200"
            />
            <div
              class="hidden lg:flex flex-col gap-y-3"
            >
              <div
                class="flex flex-col gap-3 py-3"
              >
                <div
                  class="text-dial-sapphire font-semibold"
                >
                  Bookmark
                </div>
                <button
                  class="group flex flex-row gap-x-3"
                  type="button"
                >
                  <img
                    alt="Logo of Bookmark."
                    class="object-contain"
                    height="40"
                    src="/ui/v1/bookmark-icon.svg"
                    width="40"
                  />
                  <div
                    class="my-auto border-b border-transparent group-hover:border-dial-slate-500"
                  >
                    <div
                      class="text-sm text-dial-stratos text-left"
                    >
                      Bookmark this page.
                    </div>
                  </div>
                </button>
                <div
                  class="text-sm text-dial-stratos"
                >
                  Your bookmarks can be found in your Account page.
                </div>
              </div>
              <hr
                class="border-b border-dial-slate-200"
              />
              <div
                class="flex flex-col gap-3 py-3"
              >
                <div
                  class="text-dial-sapphire font-semibold"
                >
                  Share
                </div>
                <div
                  class="flex flex-row gap-3"
                >
                  <button>
                    <img
                      alt="Logo of LinkedIn."
                      class="object-contain"
                      height="40"
                      src="/ui/v1/linkedin-icon.svg"
                      width="40"
                    />
                  </button>
                  <button>
                    <div
                      class="bg-dial-iris-blue rounded-full w-[40px] h-[40px]"
                    >
                      <div
                        class="flex items-center h-full"
                      >
                        <img
                          alt="Logo of WhatsApp."
                          class="my-auto mx-auto"
                          height="20"
                          src="/ui/v1/whatsapp-icon.svg"
                          width="20"
                        />
                      </div>
                    </div>
                  </button>
                  <button>
                    <img
                      alt="Logo of Email."
                      class="object-contain"
                      height="40"
                      src="/ui/v1/email-icon.svg"
                      width="40"
                    />
                  </button>
                  <button>
                    <img
                      alt="Logo of Copy."
                      class="object-contain"
                      height="40"
                      src="/ui/v1/copy-icon.svg"
                      width="40"
                    />
                  </button>
                </div>
              </div>
              <hr
                class="border-b border-dial-slate-200"
              />
              <div
                class="flex flex-col gap-3 py-3"
              >
                <div
                  class="text-dial-sapphire font-semibold"
                >
                  Comment
                </div>
                <div
                  class="text-sm text-dial-stratos"
                >
                  
    Have a question for the community on this Use Case Step or have some insight you would like to share?
  
                </div>
                <div
                  class="flex text-white"
                >
                  <div
                    class="bg-dial-iris-blue rounded-md text-sm"
                  >
                    <button
                      class="px-5 py-3"
                      type="button"
                    >
                      Post a comment
                    </button>
                  </div>
                </div>
              </div>
              <hr
                class="border-b border-dial-slate-200"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="lg:basis-2/3"
      >
        <div
          class="px-4 lg:px-0 py-4 lg:py-6"
        >
          <div
            class="flex flex-col gap-y-3"
          >
            <div
              class="text-xl font-semibold text-dial-blueberry py-3"
            >
              Description
            </div>
            <div
              class="block"
            >
              <div
                class="relative"
              >
                <div
                  class="html-react-parser-viewer text-base"
                >
                  Repository of @firma.
                </div>
              </div>
            </div>
            <hr
              class="border-b border-dial-blue-chalk my-3"
            />
            <div
              class="text-lg font-semibold text-dial-blueberry py-3"
            >
              Repository Location
            </div>
            <div
              class="flex pb-3 text-dial-stratos text-sm"
            >
              <a
                class="flex border-b border-dial-iris-blue "
                href="//github.com/ctt-gob-es/clienteafirma"
                rel="noreferrer"
                target="_blank"
              >
                <div
                  class="line-clamp-1 break-all"
                >
                  github.com/ctt-gob-es/clienteafirma
                </div>
              </a>
               ⧉
            </div>
            <hr
              class="border-b border-dial-blue-chalk my-3"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Unit tests for the product main page. Should render unauthorized to edit repository. 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="h-[60vh]"
  >
    <div
      class="flex bg-dial-alice-blue h-full"
    >
      <div
        class="text-dial-stratos text-lg w-full my-auto flex flex-col gap-4"
      >
        <svg
          class="spinner mx-auto"
          fill="currentColor"
          height="3em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="3em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M304 48a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zm0 416a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zM48 304a48 48 0 1 0 0-96 48 48 0 1 0 0 96zm464-48a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zM142.9 437A48 48 0 1 0 75 369.1 48 48 0 1 0 142.9 437zm0-294.2A48 48 0 1 0 75 75a48 48 0 1 0 67.9 67.9zM369.1 437A48 48 0 1 0 437 369.1 48 48 0 1 0 369.1 437z"
          />
        </svg>
        <div
          class="text-center mt-5"
        >
          Fetching data ...
        </div>
      </div>
    </div>
  </div>
</div>
`;
